localstack/services/cloudformation/resource_provider.py

```diff
@@ -809,6 +809,11 @@ def execute_action(
                         request.resource_type,
                         request.logical_resource_id,
                     )
+                    if request.previous_state is None:
+                        # this is an issue with our update detection. We should never be in this state.
+                        request.action = "Add"
```

Comment:
so is this a TODO for the framework? or what do we do about the comment?

Reply:
This is an issue with the CFn engine in LocalStack. We should never try to perform an update/modify on a resource that has no previous state, since this means the resource has not been created yet.